src/get-virtualization-type.js

```diff
@@ -0,0 +1,39 @@
+let log = require('./log');
+let debug = log.debugCompat('aws-provisioner:lib:check-for-ami');
```

Comment:
Let's not use log.debugCompat and instead use the log methods directly.

Here's some things you can do:
```javascript
log.trace({a:1}, "this is so unimportant we likely never show it");
log.debug("can log with just a string");
log.debug({b: 2}, "or also with an object.  objects let us inspect logs better");
log.info("the next log level up");
log.warn("and another level up");
log.err("here's were the good stuff is");
let anError = new Error('Sample Error');
log.error(anError, "This will do some *really* nice things like logging full stacks");
log.error({err: anError, otherField: 123}, "You can also make it the 'err' property of the object");
log.error("or just ignore");
log.fatal("this is basically just when the program is about to quit");
```

If there are time concerns around the internship timing, this is really not a huge deal, but it would be nice.  The string should always be static (no %s, %j or +'ing strings together.  The object should have enough information to debug.  Usually, this means including the region and ami set or something.